CIP-0000.md

```diff
@@ -0,0 +1,198 @@
+- - - -
+lang: en
+cip: 0000
+title: Celo Improvement Proposal (CIP) Process
+status: Draft
+type: Meta
+discussions-to:
+author: Yaz Khoury (@YazzyYaz), James Prestwich (@prestwich), Pranay Mohan (@pranaymohan), Mariano Cortesi (@mcortesi)
+created: 2020-09-22
+license: CC0 1.0
+- - - -
+
+# Abstract
+A Celo Improvement Proposal (CIP) describe standards for the Celo platform, including the core protocol specifications, SDK, and contract standards. It is a design document providing information to the Celo community, or describing a new feature for Celo or its processes or environment, a rationale for the proposal, detailed solution including technical specifications, and, if any, a list of potential risks. 
+
+The CIP proposer, the **author** is responsible for building consensus within the community and documenting whether there is rough consensus on their proposal. 
+
+Because the CIPs are maintained as text files in a versioned repository, their revision history is the historical record of the feature proposal.
+
+Participation in the Celo project and the CIP process is informed by the Celo [Community Tenets](https://celo.org/community) and subject to the  [Code of Conduct](https://celo.org/code-of-conduct).
+
+# CIP Types
+There are three types of CIP:
+
+* A **Standards Track CIP** describes any change that affects most or all Celo implementations, such as a change to the network protocol, a change in block or transaction validity rules, proposed application standards/conventions, or any change or addition that affects the interoperability of applications using Celo. Furthermore Standard CIPs can be broken down into the following categories. Standards Track CIPs consist of two parts, a design document and implementation.
+	* **Core** - improvements requiring a consensus fork, as well as changes that are not necessarily consensus critical but may be relevant to  “core dev” discussions, and the validator/node strategy.
+	* **Governance** - improvements to the governance proposal to the Celo platform including parameter changes. It should explain the reason for the change, enumerate the smart contract methods that will be called with parameters, and describe the impact and any risks. 
+	* **Networking** - improvements to networking protocol specifications such as devp2p, and proposed improvements like whisper and swarm.
```

Comment:
Assuming this was inherited from EIPs? How relevant is this to Celo given the current usage of whisper and swam?

Reply:
I've listed those as examples inherited from EIPs. I can remove them if needed for clarity. For swarm, EF recently released this: https://swarm.ethereum.org/

Whisper, I've seen working in the wild between different networks. It can be useful to send messages from one node to others.

Reply:
Devp2p certainly is part of this

Reply:
Ah I guess I meant the usage of whisper/swarm in Celo.

Good to know that whisper more mature, we looked into this a couple years ago and couldn't find compelling examples of it being used in the wild

Reply:
Resolving since we have moved to a Ring level categorization instead of a core/non-core one.